virtwho/virt/rhevm/rhevm.py

```diff
@@ -71,18 +71,73 @@ def __init__(self, logger, config):
         self.username = self.config.username
         self.password = self.config.password
 
-        self.clusters_url = urlparse.urljoin(self.url, "api/clusters")
-        self.hosts_url = urlparse.urljoin(self.url, "api/hosts")
-        self.vms_url = urlparse.urljoin(self.url, "api/vms")
-
         self.auth = HTTPBasicAuth(self.config.username, self.config.password)
 
+    @property
+    def clusters_url(self):
+        if not hasattr(self, '_clusters_url'):
+            if self.major_version == '4' and 'ovirt-engine' not in self.url:
```

Comment:
I'd like to request that we move the logic of determining the full url for access to a particular collection into it's own method. In this way should there ever be a change as to how we determine what the url should be, we need only change it in one place.

It might also make sense to set the value of each of these attributes ({clusters,hosts,vms}_url) in the prepare method.